src/controls.c

```diff
@@ -0,0 +1,100 @@
+#include "../include/controls.h"
+#include "SDL.h"
+#include <stdio.h>
+
+// Function definitions
+void init_emulator(State *state)
+{
+    // Initialize all necessary parts of the state
+    state->a = 0;
+    state->pc = 0;
+    state->input_state = 0;
+    // Initialize other parts of the state as needed
+}
+
+// Emulate 8080 processor (stub function)
+void emulate8080(State *state)
+{
+    // Implement the 8080 emulation logic here
+}
+
+// Update input state based on SDL events
+void update_input_state(State *state, SDL_Event *e)
```

Comment:
maybe call this get_keyboard_input or something since we won't have an input_state